constellations/gemini-user/src/account/account.js

```diff
@@ -0,0 +1,658 @@
+                                (element) => element.id == afh.afhId
+                            );
+                            upcomingAFHs.push(matchedAFH);
+                        });
+                        upcomingAFHs = union(upcomingAFHs);
+                        this.setState({ upcomingAFHs: upcomingAFHs });
+                    });
+                });
+            })
+            .catch((err) => alert("Could not fetch data: " + err));
+        API.get("api/transactions/account/" + id).then((res) =>
+            this.setState({ transactions: res.data })
+        );
+    };
+
+    toggleViewAllClasses = () => {
+        this.setState({
+            viewAllEnrolledClasses: !this.state.viewAllEnrolledClasses,
+        });
+    };
+
+    onTabSelect = (tab) => {
+        this.setState({ selectedTab: tab.toLowerCase() });
+    };
+
+    onPasswordChange = (password) => {
+        this.setState({ password: password });
+    };
+
+    render = () => {
+        const tabButtons = ["Settings", "Registrations", "Payment"].map(
+            (item, index) => {
+                return (
+                    <div
+                        className={
+                            this.state.selectedTab == item.toLowerCase()
+                                ? "selected"
+                                : ""
+                        }
+                        onClick={(e) => this.onTabSelect(item)}
+                        key={index}>
+                        {item}
+                    </div>
+                );
+            }
+        );
+
+        let tabContent;
+        switch (this.state.selectedTab) {
+            case "settings":
+                tabContent = (
+                    <SettingsTab
+                        users={this.state.users}
+                        primaryEmail={this.state.primaryEmail}
+                        accountId={this.state.id}
+                        password={this.state.password}
+                        passwordChangeCallback={this.onPasswordChange}
+                    />
+                );
+                break;
+            case "registrations":
+                tabContent = this.state.viewAllEnrolledClasses ? (
+                    <RegistrationsTabAllClasses
+                        userClasses={this.state.userClasses}
+                        toggleTabCallback={this.toggleViewAllClasses}
+                    />
+                ) : (
+                    <RegistrationsTabMain
+                        userClasses={this.state.userClasses}
+                        upcomingAFHs={this.state.upcomingAFHs}
+                        locationsById={this.state.locationsById}
+                        toggleTabCallback={this.toggleViewAllClasses}
+                    />
+                );
+                break;
+            case "payment":
+                tabContent = (
+                    <PaymentTab transactions={this.state.transactions} />
+                );
+                break;
+        }
```

Comment:
Ok sweet, this is pretty well done actually. My only gripe is this one javascript file is super large for what it can be.
I think it would be better if we had a couple of files instead of one large file. And that way, each component can keep better track of their own isolated content.

So have a `src/account/settings.js`, `src/account/registrations.js`, `src/account/payments.js` files containing the SettingsTab, RegistrationXTab, and PaymentTab respectively. And this file `account.js` will import components from those 3 files. Now each file is only 150-250 lines and much easier to see what's being managed by what.

For example, transactions are only used in the PaymentTab. So technically, PaymentTab can be its own component and its componentDidMount() function can fetch the transactions. So now AccountPage doesn't have to worry about transactions, it just has to manage when to show PaymentTab.

So now all transactions stuff can be removed from AccountPage and placed into PaymentTab. In order to get all transactions for an account, PaymentTab needs an `accountId`, which `AccountPage` can simply just pass as a parameter to PaymentTab.

```
let tabContent = <PaymentTab acccountId={this.state.accountId}/>
```

Inside PaymentTab, we fetch the transactions, and display what we need to display. The same can be said for the other Tab components. They all have information only they care about, so let those components handle those by themselves. Have AccountPage only manage which tab to show to the user.